src/Classes/ServiceAPI/MyRadio_TracklistItem.php

```diff
@@ -70,6 +71,119 @@ protected static function factory($id)
         return new self($result);
     }
 
+
+    /**
+     * Create a new TracklistItem, returning the new item.
+     *
+     * @param int $trackid  The ID of the track to tracklist.
+     * @param int $timeslotid   The ID of the timeslot to tracklist to. Optional, defaults to current show.
+     * @param int $starttime    Epoch time of the start of the tracklist. Optional, defaults to current time.
+     * @param char $sourceid    The id of the tracklist source (baps, webstudio, etc), see tracklist.source. Defaults to 'api'
+     * @param char $state       The state of the tracklist, see tracklist.state. Defaults to 'confirmed'
+     *
+     *
+     * @return MyRadio_TracklistItem
+     *
+     * @throws MyRadioException
+     */
+    public static function create($trackid, $timeslotid = null, $starttime = null, $sourceid = 'a', $state = 'c')
+    {
+
+        if (AuthUtils::hasPermission(AUTH_TRACKLIST_ALL)) {
+            $tracklist_all = true;
+        }
+        else if (AuthUtils::hasPermission(AUTH_TRACKLIST_OWN))
+            $tracklist_all = false;
+        else {
+            throw new MyRadioException("The current user does not have permission to create a tracklistitem.", 403);
+        }
+
+        if ($timeslotid != null && $tracklist_all == false) {
+            throw new MyRadioException("The current user doesn't have permission to set a tracklist on a show other than their own.", 403);
+        }
+
+        if ($timeslotid == null) {
+            $timeslot = MyRadio_Timeslot::getCurrentTimeslot();
+            $timeslotid = $timeslot != null ? $timeslot->getID() : null; // will be null if jukebox etc.
+        } else {
+            $timeslot = MyRadio_Timeslot::getInstance($timeslotid);
+        }
+
+        if ($timeslot == null) {
+            // we're on jukebox
+            if ($tracklist_all == false) {
+                throw new MyRadioException("The current user doesn't have permission to set a tracklist on a show other than their own.", 403);
+            }
+        } else {
+            if ($tracklist_all == false && !in_array(MyRadio_User::getCurrentUser(), $timeslot->getSeason()->getShow()->getCreditObjects())) {
```

Comment:
Are `MyRadio_User` objects singletons? (that is, is it guaranteed that the object for user 123 in `getCreditObjects()` and the object that `getCurrentUser()` returns will be the same?